lib/Raven/Breadcrumbs/ErrorHandler.php

```diff
@@ -21,16 +21,11 @@ public function __construct(\Raven\Client $ravenClient)
 
     public function handleError($code, $message, $file = '', $line = 0, $context = array())
     {
-        $this->ravenClient->breadcrumbs->record(array(
-            'category' => 'error_reporting',
-            'message' => $message,
-            'level' => $this->ravenClient->translateSeverity($code),
-            'data' => array(
-                'code' => $code,
-                'line' => $line,
-                'file' => $file,
-            ),
-        ));
+        $this->ravenClient->leaveBreadcrumb($this->ravenClient->translateSeverity($code), Breadcrumb::TYPE_ERROR, 'error_reporting', $message, [
```

Comment:
It's still not clear to me what are exactly the concerns on this PR. This is what I've done and why, in the hope that this is enough to demonstrate that it's not something related to personal preferences, but to make the code better:

- Breadcrumbs are not accessible anymore after being stored outside of the client instance: although this can be changed just by adding a getter to the `Client` instance, I've done this because I thought that there are no reasons I can think of that a user would want to change a breadcrumb after having added it
- The function to add a breadcrumb does not accept anymore an array of data: I've done this because having explicit function arguments means that parameters are clear and IDE provides autocomplete while before you did not know what were the arguments. You had to know that you could pass a `message` key to specify the text of the breadcrumb or a `data` key to specify additional metadata.
- The breadcrumbs are stored as objects instead of arrays: this leads to better encapsulation of the data, code is clearer and most importantly objects are passed by reference but array aren't, and we use breadcrumbs enough time to think of this as a performance optimization
- The `Recorder` class is basically the old `Breadcrumbs` class, renamed to something that makes clear what it really does and that better encapsulates that logic. It also implements an interface, so users can switch implementation easily

My doubts are about the function to record a breadcrumb. Maybe it would be better to accept an instance of the `Breadcrumb` class instead of having a set of arguments. This way users will have to create the instance themselves, but they can use the setters to set the data and it would still be more clear than having to know the structure of an array.

I also had a look at the other SDKs. I don't know neither Python nor Ruby, but as far as I see Python has named arguments while the Ruby SDK imho has the same problem of this one, which is that end users have to know which keys can go into the input array